src/cloud-sql-instance.ts

```diff
@@ -49,23 +51,33 @@ export class CloudSQLInstance {
   }
 
   private readonly connectionType: IpAdressesTypes;
+  private readonly authType: AuthTypes;
   private readonly sqlAdminFetcher: Fetcher;
   private refreshTimeoutID?: ReturnType<typeof setTimeout>;
   private closed = false;
   public readonly instanceInfo: InstanceConnectionInfo;
+  public auth?: GoogleAuth;
   public ephemeralCert?: SslCert;
   public host?: string;
   public privateKey?: string;
   public serverCaCert?: SslCert;
 
   constructor({
     connectionType,
+    authType,
     instanceConnectionName,
     sqlAdminFetcher,
   }: CloudSQLInstanceOptions) {
     this.connectionType = connectionType;
+    this.authType = authType;
     this.instanceInfo = parseInstanceConnectionName(instanceConnectionName);
     this.sqlAdminFetcher = sqlAdminFetcher;
+
+    if (authType === AuthTypes.IAM) {
+      this.auth = new GoogleAuth({
+        scopes: ["https://www.googleapis.com/auth/sqlservice.login"],
+      });
+    }
```

Comment:
Do we really need an independent `GoogleAuth` instance? Looking at where it's used in `src/sqladmin-fetcher.ts` it seems to me that this new instance is only being used to call `auth.getAccessToken()`.

If that's the case it seems much simpler to [keep a reference to the `GoogleAuth` instance declared in the constructor of the `SQLAdminFetcher` class](https://github.com/GoogleCloudPlatform/cloud-sql-nodejs-connector/blob/93168049995520bc290368b0530fd768d052db38/src/sqladmin-fetcher.ts#L33) and simply call that method in that same instance on `getEphemeralCertificate`. Which should avoid all changes to `src/cloud-sql-instance.ts`.

Reply:
So the reason we're creating a new Auth instance here is that this instance has more restricted scopes (only `sqlservice.login`, while the one in `src/sqladmin-fetcher.ts` has `sqlservice.admin`

Reply:
I see! that makes sense, I still wonder though if it wouldn't be simpler than to have a single extra `GoogleAuth` instance living in `src/sqladmin-fetcher.ts` (something like `this.iamAuth`) instead of adding a new `auth` reference for each cloud sql instance

Reply:
I think that's a good idea, as long as we'll never run into a case where we'd need to use different auth sources with the same connector. I don't think it will be a problem though.

Reply:
cool 👍 yeah, let's move it there then, I'm not really liking the idea of adding `GoogleAuth` into `src/cloud-sql-instance.ts`

Reply:
We may want to discuss this before implementing it for feature parity across connectors. As I know atleast for the Python Connector we currently attach a Google Auth credentials object to each instance at the instance level. It might be a good enhancement across the board to move the Google Auth credentials up a level to the Connector level but we should probably discuss it a bit more to make sure there isn't a use case for leaving it per instance.

Reply:
Java is also at the instance level (because before the refactor there wasn't a connector level). But I agree, let's leave it as-is for now and discuss this more.

Reply:
Having in mind that the go connector is the reference implementation it seems to me like an additional argument to move it out of the instance level.